core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestExceptionMapper.java

```diff
@@ -33,13 +33,13 @@ public class S3RestExceptionMapper implements ExceptionMapper<Throwable> {
   @Override
   public Response toResponse(Throwable e) {
     if (e instanceof AlluxioStatusException) {
-      return S3RestUtils.createErrorResponse((AlluxioStatusException) e);
+      return S3RestUtils.createErrorResponse(null, (AlluxioStatusException) e);
     } else if (e instanceof AlluxioRuntimeException) {
-      return S3RestUtils.createErrorResponse((AlluxioRuntimeException) e);
+      return S3RestUtils.createErrorResponse(null, (AlluxioRuntimeException) e);
     } else if (e instanceof S3Exception) {
       return S3RestUtils.createErrorResponse((S3Exception) e);
     } else if (e instanceof IOException) {
-      return S3RestUtils.createErrorResponse((IOException) e);
+      return S3RestUtils.createErrorResponse(null, (IOException) e);
     } else {
       return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
           .entity(e.getMessage()).build();
```

Comment:
Is it possible to simplify this? Like by moving all those individual handlers to be `private` in `S3RestUtils`, and creating a single `public` method which does the `instanceof` checks?

```
public static Response createErrorResponse(Exception e) {
  return createErrorResponse(S3Constants.EXCEPTION_MAPPER_RESOURCE, e);
}

public static Response createErrorResponse(String resource, Exception e) {
  if (e instanceof AlluxioStatusException) {
    return createErrorResponse(resource, (AlluxioStatusException) e);
  }
  ...
  if (e instanceof IOException) {
    return createErrorResponse(resource, (IOException) e);
  }
  return createErrorResponse(resource, e);  // generic `Exception` signature
}

private static Response createErrorResponse(String resource, AlluxioStatusException e) {
  ...
}
```